modulo2/projeto-labefy/projeto-labefy/src/Components/Playlist.js

```diff
@@ -0,0 +1,215 @@
+    align-items: center;
+    padding-top: 20px;
+    padding-bottom: 20px;
+
+    button{
+        height: 25px;
+    }
+`
+
+const DivTracks = styled.div`
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: space-between;
+
+    li{
+        margin-right: 10px;
+        margin-bottom: 10px;
+    }
+
+    button{
+        margin-bottom: 10px;
+    }
+`
+
+const DivInputSong = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    margin-top: 15px;
+`
+
+export default class Playlist extends React.Component{
+
+    state = {
+        playlist: [],
+        tracks: [],
+        name: "",
+        id: "",
+        artist: "",
+        url: "",
+        addSong: 0,
+    }
+
+    componentDidMount() {
+        this.getAllPlaylist()
+    }
+
+    componentDidUpdate(){
+        this.getAllPlaylist()
+    }
+
+    handleName = (e) => {
+        this.setState({name: e.target.value})
+    }
+
+    handleArtist = (e) => {
+        this.setState({artist: e.target.value})
+    }
+
+    handleUrl = (e) => {
+        this.setState({url: e.target.value})
+    }
+
+
+    toAddSong = () => {
+        if(this.state.addSong === 0){
+            return this.setState({addSong: 1})
+        } else{
+            return this.setState({addSong: 0})
+        }
+    }
+
+    ShowInputAddSong = () => {
+        
+        if(this.state.addSong === 1){
+            return(
+            <DivInputSong>
+                <input placeholder="Insert the new song" value={this.state.name} onChange={this.handleName}></input>
+                <input placeholder="Insert the artist" value={this.state.artist} onChange={this.handleArtist}></input>
+                <input placeholder="Insert the url" value={this.state.url} onChange={this.handleUrl}></input>
+                <button onClick={() => this.addtrackToPlaylist()}>Save</button>
+            </DivInputSong>
+            )
+        } else{
+            return ""
+        } 
+    }
+
+    getAllPlaylist = async () =>{
+        const url = 'https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists'
+        const headers = {headers : {
+                Authorization: "andre-marques-carver"
+            }
+        }    
+        
+        try{
+            const response = await axios.get(url, headers)
+            this.setState({playlist: response.data.result.list})
+        } catch(err) {
+            console.log(err.response)
+        }
+    }
+
+    getPlaylistTrack = async (id) => {
+        const url = `https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${id}/tracks`
+        const headers ={ headers :{
+                Authorization: "andre-marques-carver"
+            }
+        }
+
+        try{
+            const response = await axios.get(url, headers)
+            console.log(response.data.result.tracks)
+            this.setState({tracks: response.data.result.tracks})
+            this.setState({id: id})
```

Comment:
Forma correta de usar o this.setState nessa situação é fazer tudo de uma vez.
exemplo
```
this.setState({
  tracks: response.data.result.tracks,
  id: id
})
```